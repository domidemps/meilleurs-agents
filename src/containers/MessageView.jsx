/** @jsx jsx */
import {css, jsx} from '@emotion/core'
import Paper from '@material-ui/core/Paper'
import {useSelector} from 'react-redux'
import Typography from '@material-ui/core/Typography'
import {lighten} from '@material-ui/core'
import dayjs from 'dayjs'

import {TEXT_COLOR, THIRD_COLOR} from '../styles/material_ui_raw_theme_file'

const styles = css`
  width: 70%;
  display: flex;
  flex-direction: column;
  .papers {
    padding: 20px;
    margin: 15px;
    color: ${TEXT_COLOR};
  }
  .paperTop {
    height: 15%;
  }
  .paperBottom {
    height: 85%;
  }
  .infos {
    display: flex;
    flex-direction: row;
    margin-top: 10px;
  }
  .contactLabel {
    color: ${lighten(TEXT_COLOR, 0.3)};
    margin: 10px 0px 0px 10px;
  }
  .contactValue {
    color: ${THIRD_COLOR};
    margin: 10px 0px 0px 50px;
  }
`

export default function MessageView() {
  const selectedMessage = useSelector(state => state.messages.selectedMessage)

  if (!Boolean(selectedMessage)) {
    return <div css={styles} />
  }

  const renderContactInformation = messageContact => {
    return (
      <Paper className="papers paperTop">
        <Typography
          variant="h5"
          key="sender"
          css={css`
            font-weight: 600;
          `}>
          {messageContact.firstname} {messageContact.lastname}
        </Typography>
        <div className="infos">
          <div key="labels">
            <Typography variant="body1" key="email" className="contactLabel">
              Email
            </Typography>
            <Typography variant="body1" key="phone" className="contactLabel">
              Téléphone
            </Typography>
          </div>
          <div key="values" className="inLine">
            <Typography variant="body1" key="email" className="contactValue">
              {messageContact.email}
            </Typography>
            <Typography variant="body1" key="phone" className="contactValue">
              {messageContact.phone.match(/.{1,2}/g).join(' ')}
            </Typography>
          </div>
        </div>
      </Paper>
    )
  }

  const renderMessageContent = message => {
    return (
      <Paper className="papers paperBottom">
        <Typography
          variant="h5"
          key="sender"
          css={css`
            font-weight: 600;
          `}>
          {message.contact.firstname} {message.contact.lastname}
        </Typography>
        <Typography
          variant="h6"
          key="dateTime"
          css={css`
            color: ${lighten(TEXT_COLOR, 0.5)};
          `}>
          Le {dayjs(message.date).format('DD/MM/YYYY')} à {dayjs(message.date).format('HH:mm')}
        </Typography>
        <div
          css={css`
            margin-top: 30px;
            line-height: 20px;
          `}>
          {message.body}
        </div>
      </Paper>
    )
  }

  return (
    <div css={styles}>
      {renderContactInformation(selectedMessage.contact)}
      {renderMessageContent(selectedMessage)}
    </div>
  )
}
