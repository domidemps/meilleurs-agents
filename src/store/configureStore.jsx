import {applyMiddleware, compose, createStore} from 'redux'
import {persistReducer, persistStore} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import {routerMiddleware} from 'connected-react-router'
import promise from 'redux-promise'
import thunk from 'redux-thunk'

import reducers, {history} from '../reducers'

const middlewares = [thunk, promise, routerMiddleware(history)]

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['realtors', 'messages'],
}

const persistedRootReducer = persistReducer(persistConfig, reducers)
const createStoreWithEnhancers = composeEnhancers(applyMiddleware(...middlewares))(createStore)
export const store = createStoreWithEnhancers(persistedRootReducer)
export const persistor = persistStore(store)
