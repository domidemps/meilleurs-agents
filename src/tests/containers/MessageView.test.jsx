import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'
import {createMemoryHistory} from 'history'
import {Router, Switch} from 'react-router-dom'

import MessageView from '../../containers/MessageView'

export const history = createMemoryHistory({initialEntries: ['/']})

describe('MessageView', () => {
  const store = configureStore([thunk])({
    realtors: {
      selectedAgency: '101',
    },
    messages: {
      selectedMessage: {
        body:
          "Lorem Ipsum #10156 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
        contact: {
          email: 'jbrown@gmail.com',
          firstname: 'John',
          lastname: 'Brown',
          phone: '0671459784',
        },
        date: '2020-11-23T19:58:02.673783',
        id: 10156,
        read: true,
        subject: 'Email #10156',
        type: 'email',
      },
    },
  })
  it('renders correctly for computers/tablets', () => {
    global.innerWidth = 600
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router history={history}>
              <Switch>
                <MessageView />
              </Switch>
            </Router>
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot()
  })
  it('renders correctly for smartphones', () => {
    global.innerWidth = 350
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router history={history}>
              <Switch>
                <MessageView />
              </Switch>
            </Router>
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot()
  })
})
