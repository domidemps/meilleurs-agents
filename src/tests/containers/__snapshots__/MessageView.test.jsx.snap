// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageView renders correctly for computers/tablets 1`] = `
<div
  className="css-12yxprc-MessageView"
>
  <div
    className="MuiPaper-root papers paperTop MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h5
      className="MuiTypography-root css-5kj36g MuiTypography-h5"
    >
      John
       
      Brown
    </h5>
    <div
      className="infos"
    >
      <div>
        <p
          className="MuiTypography-root contactLabel MuiTypography-body1"
        >
          Email
        </p>
        <p
          className="MuiTypography-root contactLabel MuiTypography-body1"
        >
          Téléphone
        </p>
      </div>
      <div
        className="inLine"
      >
        <p
          className="MuiTypography-root contactValue MuiTypography-body1"
        >
          jbrown@gmail.com
        </p>
        <p
          className="MuiTypography-root contactValue MuiTypography-body1"
        >
          06 71 45 97 84
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root papers paperBottom MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h5
      className="MuiTypography-root css-5kj36g MuiTypography-h5"
    >
      John
       
      Brown
    </h5>
    <h6
      className="MuiTypography-root css-wc552f MuiTypography-h6"
    >
      Le 
      23/11/2020
       à 
      19:58
    </h6>
    <div
      className="messageBody"
    >
      Lorem Ipsum #10156 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </div>
  </div>
</div>
`;

exports[`MessageView renders correctly for smartphones 1`] = `
<div
  className="css-12yxprc-MessageView"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained css-1stnv2n-MessageView MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
          />
        </svg>
      </span>
      Retour
    </span>
  </button>
  <div
    className="MuiPaper-root papers paperTop MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h5
      className="MuiTypography-root css-5kj36g MuiTypography-h5"
    >
      John
       
      Brown
    </h5>
    <div
      className="infos"
    >
      <div>
        <p
          className="MuiTypography-root contactLabel MuiTypography-body1"
        >
          Email
        </p>
        <p
          className="MuiTypography-root contactLabel MuiTypography-body1"
        >
          Téléphone
        </p>
      </div>
      <div
        className="inLine"
      >
        <p
          className="MuiTypography-root contactValue MuiTypography-body1"
        >
          jbrown@gmail.com
        </p>
        <p
          className="MuiTypography-root contactValue MuiTypography-body1"
        >
          06 71 45 97 84
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root papers paperBottom MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h5
      className="MuiTypography-root css-5kj36g MuiTypography-h5"
    >
      John
       
      Brown
    </h5>
    <h6
      className="MuiTypography-root css-wc552f MuiTypography-h6"
    >
      Le 
      23/11/2020
       à 
      19:58
    </h6>
    <div
      className="messageBody"
    >
      Lorem Ipsum #10156 is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </div>
  </div>
</div>
`;
